src/test/java/teammates/test/cases/common/FieldValidatorTest.java

```diff
@@ -393,76 +393,89 @@ private void invalidityInfoFor_invalidGender_shouldReturnErrorString() {
     }
 
     @Test
-    public void testGetValidityInfo_GOOGLE_ID() {
-        
-        verifyAssertError("null value", FieldType.GOOGLE_ID, null);
-        verifyAssertError("contains '@gmail.com'", FieldType.GOOGLE_ID, "abc@GMAIL.com");
-        
-        
-        testOnce("valid: typical value", 
-                FieldType.GOOGLE_ID, 
-                "valid9.Goo-gle.id_", 
-                "");
-        
-        testOnce("valid: minimal non-email id accepted", 
-                FieldType.GOOGLE_ID, 
-                "e", 
-                "");
-        
-        testOnce("valid: typical email address used as id", 
-                FieldType.GOOGLE_ID, 
-                "someone@yahoo.com", "");
-        
-        testOnce("valid: minimal email id accepted as id", 
-                FieldType.GOOGLE_ID, 
-                "e@y", 
-                "");
-        
-        String maxLengthValue = StringHelper.generateStringOfLength(GOOGLE_ID_MAX_LENGTH);
-        testOnce("valid: max length", 
-                FieldType.GOOGLE_ID, 
-                maxLengthValue, 
-                "");
+    public void testGetInvalidityInfoFor_GoogleId() {
+        invalidityInfoFor_nullGoogleId_throwException();
+        invalidityInfoFor_googleIdWithGmailDomain_throwException();
+        invalidityInfoFor_validGoogleId_returnEmptyString();
+        invalidityInfoFor_invalidGoogleId_returnErrorString();
+    }
```

Comment:
@damithc I was thinking it would be neater to group related tests (i.e., for the same type of field) together.

Nonetheless, I just consulted the testing best practices doc and it suggests having many small tests. I think smaller and more specific tests are better too.

I'll split the tests accordingly (and also rename them to maintain the logical grouping).